### DISCUSSION TITLE
is the container work on Mac OS docker?

### DISCUSSION THREAD

**Discussion by fllay** [herlesupreeth/docker_open5gs#250] [Q&A]
I am trying to set up 5g SA core network on Mac. It seems to run ok but when I tried to connect gnb from other machine it said 
```
Connecting to AMF on 192.168.68.10:38412
Failed to connect SCTP socket to 192.168.68.10:38412. error="Connection timed out" timeout=1385ms
srsGNB ERROR: Failed to create SCTP gateway.
```
However, the web interface is fine on that 192.168.68.10 address.  I can add subscriber just fine. Any idea what is wrong?

**Comment by herlesupreeth** [comment]
The main reason why it doesnt work in MAC OS is due to lack of kernel based SCTP library. 
If you are just gonna use 5G Core then its still possible to run on MAC OS by using usrsctp but for that you need to change compilation steps of open5gs

**Reply by herlesupreeth** [reply]
Ofcourse, you cant run SRS gNB/eNB on native MAC OS since there is no alternative SCTP library being supported.

**Reply by fllay** [reply]
Could you clarify which part of open5gs build I need to change? I saw `libusrsctp` installation on build instruction for Mac OS but how can I use that in the Docker? Thank you

**Reply by herlesupreeth** [reply]
You can check this commit for the modification to be done -https://github.com/open5gs/open5gs/commit/357d84abb4145b060f1c4fc49bbfaee1c56929cf

**Reply by herlesupreeth** [reply]
In order to use that in docker you would need to either fork open5gs and apply the diff like I did in that commit I pasted above and change the github url in base/Dockerfile
